From 6d62fabd2577150a9c880f8893b5e35c7a025252 Mon Sep 17 00:00:00 2001
From: Tomas Winkler <tomas.winkler@intel.com>
Date: Wed, 30 Jul 2014 17:10:59 +0300
Subject: [PATCH] mei: txe: add cherrytrail device id

Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
---
 drivers/misc/mei/pci-txe.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/misc/mei/pci-txe.c b/drivers/misc/mei/pci-txe.c
index 07797df..88a3d2b 100644
--- a/drivers/misc/mei/pci-txe.c
+++ b/drivers/misc/mei/pci-txe.c
@@ -37,6 +37,7 @@
 
 static const struct pci_device_id mei_txe_pci_tbl[] = {
 	{MEI_PCI_DEVICE(0x0F18, mei_txe_cfg)}, /* Baytrail */
+	{MEI_PCI_DEVICE(0x2298, mei_txe_cfg)}, /* Cherrytrail */
 	{0, }
 };
 MODULE_DEVICE_TABLE(pci, mei_txe_pci_tbl);
-- 
1.7.9.5

