From 08cba1e4482b880e2d6706e6c6ac4e1093f4c3f6 Mon Sep 17 00:00:00 2001
From: Jianxun Zhang <jianxun.zhang@intel.com>
Date: Sun, 25 May 2014 17:54:13 -0700
Subject: [PATCH 2/2] Bluetooth: Add RT8723 chip ID in rfkill-gpio driver

Signed-off-by: Loic Poulain <loic.poulain@intel.com>
Signed-off-by: Jianxun Zhang <jianxun.zhang@intel.com>
---
 net/rfkill/rfkill-gpio.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/rfkill/rfkill-gpio.c b/net/rfkill/rfkill-gpio.c
index 9137658771a3..73299df888fe 100644
--- a/net/rfkill/rfkill-gpio.c
+++ b/net/rfkill/rfkill-gpio.c
@@ -169,10 +169,12 @@ static const struct acpi_device_id rfkill_acpi_match[] = {
 	{ "BCM2E1A", RFKILL_TYPE_BLUETOOTH },
 	{ "BCM2E39", RFKILL_TYPE_BLUETOOTH },
 	{ "BCM2E3D", RFKILL_TYPE_BLUETOOTH },
+	{ "OBDA8723", RFKILL_TYPE_BLUETOOTH },
 	{ "BCM4752", RFKILL_TYPE_GPS },
 	{ "LNV4752", RFKILL_TYPE_GPS },
 	{ },
 };
+MODULE_DEVICE_TABLE(acpi, rfkill_acpi_match);
 #endif

 static struct platform_driver rfkill_gpio_driver = {
--
1.9.3
