From f14cae71186cac943644f30c53984c037389f663 Mon Sep 17 00:00:00 2001
From: Aubrey Li <aubrey.li@intel.com>
Date: Fri, 9 May 2014 01:25:49 -0500
Subject: [PATCH 092/152] Support low power S0idle via PM_SUSPEND_MEM

Signed-off-by: Aubrey Li <aubrey.li@intel.com>
---
 drivers/acpi/sleep.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/acpi/sleep.c b/drivers/acpi/sleep.c
index 0fe057c..90a0efa 100644
--- a/drivers/acpi/sleep.c
+++ b/drivers/acpi/sleep.c
@@ -580,10 +580,16 @@ static int acpi_suspend_state_valid(suspend_state_t pm_state)
 	switch (pm_state) {
 	case PM_SUSPEND_ON:
 	case PM_SUSPEND_STANDBY:
-	case PM_SUSPEND_MEM:
 		acpi_state = acpi_suspend_states[pm_state];
 
 		return sleep_states[acpi_state];
+	case PM_SUSPEND_MEM:
+		acpi_state = acpi_suspend_states[pm_state];
+		/*
+		 * S3 is supported via PM_SUSPEND_MEM,
+		 * as well as low power S0idle
+		 */
+		return sleep_states[acpi_state] || low_power_s0idle;
 	default:
 		return 0;
 	}
-- 
1.7.10.4

