From 8c6e155f821236aaffaec00c4efde7301d0d54aa Mon Sep 17 00:00:00 2001
From: Aubrey Li <aubrey.li@intel.com>
Date: Sat, 10 May 2014 13:10:08 -0500
Subject: [PATCH 094/152] FIXME: coding hack to put non-boot CPUs enter C7
 with hint = 0x64

Signed-off-by: Aubrey Li <aubrey.li@intel.com>
---
 arch/x86/kernel/smpboot.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/arch/x86/kernel/smpboot.c b/arch/x86/kernel/smpboot.c
index a32da80..82b2710 100644
--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -1402,8 +1402,7 @@ static inline void mwait_play_dead(void)
 				highest_subcstate = edx & MWAIT_SUBSTATE_MASK;
 			}
 		}
-		eax = (highest_cstate << MWAIT_SUBSTATE_SIZE) |
-			(highest_subcstate - 1);
+		eax = (highest_cstate << MWAIT_SUBSTATE_SIZE) | 4;
 	}
 
 	/*
@@ -1412,7 +1411,7 @@ static inline void mwait_play_dead(void)
 	 * content is immaterial as it is not actually modified in any way.
 	 */
 	mwait_ptr = &current_thread_info()->flags;
-
+	pr_info("cpu[%d] mwait hint: 0x%x\n", smp_processor_id(), eax);
 	wbinvd();
 
 	while (1) {
-- 
1.7.10.4

