From 6e4ce6b5eac277fec1b072d35a31e67d3b170eeb Mon Sep 17 00:00:00 2001
From: Dmitry Shmidt <dimitrysh@google.com>
Date: Wed, 29 Jul 2009 10:22:03 -0700
Subject: [PATCH 045/634] mmc: sdio: Claim host in sdio_reset_comm()

Signed-off-by: Dmitry Shmidt <dimitrysh@google.com>
---
 drivers/mmc/core/sdio.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/core/sdio.c b/drivers/mmc/core/sdio.c
index 0032e0e..c335b4a 100644
--- a/drivers/mmc/core/sdio.c
+++ b/drivers/mmc/core/sdio.c
@@ -10,6 +10,7 @@
  */
 
 #include <linux/err.h>
+#include <linux/module.h>
 #include <linux/pm_runtime.h>
 
 #include <linux/mmc/host.h>
@@ -1281,6 +1282,8 @@ int sdio_reset_comm(struct mmc_card *card)
 	int err;
 
 	printk("%s():\n", __func__);
+	mmc_claim_host(host);
+
 	mmc_go_idle(host);
 
 	mmc_set_clock(host, host->f_min);
@@ -1321,13 +1324,12 @@ int sdio_reset_comm(struct mmc_card *card)
 	err = sdio_enable_wide(card);
 	if (err)
 		goto err;
-
+	mmc_release_host(host);
 	return 0;
- err:
+err:
 	printk("%s: Error resetting SDIO communications (%d)\n",
 	       mmc_hostname(host), err);
+	mmc_release_host(host);
 	return err;
 }
 EXPORT_SYMBOL(sdio_reset_comm);
-
-
-- 
1.9.1

