From feed9e8a639e6e81ead89e19f38a5304b0dd5969 Mon Sep 17 00:00:00 2001
From: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Date: Wed, 4 Dec 2013 11:20:27 +0000
Subject: [PATCH 061/351] ASoC: florida: hook up trace capture streams

Change-Id: I1a99f20b2603a182c2217805c56d18e3b9c73401
Signed-off-by: Richard Fitzgerald <rf@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/codecs/florida.c |   26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index d7cea71..2c0bb1a 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -1345,6 +1345,11 @@ static const struct snd_soc_dapm_route florida_dapm_routes[] = {
 	{ "Voice Control CPU", NULL, "SYSCLK" },
 	{ "Voice Control DSP", NULL, "SYSCLK" },
 
+	{ "Trace CPU", NULL, "Trace DSP" },
+	{ "Trace DSP", NULL, "DSP1" },
+	{ "Trace CPU", NULL, "SYSCLK" },
+	{ "Trace DSP", NULL, "SYSCLK" },
+
 	{ "IN1L PGA", NULL, "IN1L" },
 	{ "IN1R PGA", NULL, "IN1R" },
 
@@ -1678,6 +1683,27 @@ static struct snd_soc_dai_driver florida_dai[] = {
 			.formats = FLORIDA_FORMATS,
 		},
 	},
+	{
+		.name = "florida-cpu-trace",
+		.capture = {
+			.stream_name = "Trace CPU",
+			.channels_min = 2,
+			.channels_max = 8,
+			.rates = FLORIDA_RATES,
+			.formats = FLORIDA_FORMATS,
+		},
+		.compress_dai = 1,
+	},
+	{
+		.name = "florida-dsp-trace",
+		.capture = {
+			.stream_name = "Trace DSP",
+			.channels_min = 2,
+			.channels_max = 8,
+			.rates = FLORIDA_RATES,
+			.formats = FLORIDA_FORMATS,
+		},
+	},
 };
 
 static irqreturn_t adsp2_irq(int irq, void *data)
-- 
1.7.9.5

