From 9cf17fffc8fa5ba8e53b8dece7fcfac7873aab2c Mon Sep 17 00:00:00 2001
From: Dmitry Shmidt <dimitrysh@google.com>
Date: Mon, 30 Aug 2010 10:29:37 -0700
Subject: [PATCH 067/634] net: Fix CONFIG_RPS option to be turned off

Signed-off-by: Dmitry Shmidt <dimitrysh@google.com>
---
 net/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/Kconfig b/net/Kconfig
index 6cca0ad88..219a8c1 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -237,7 +237,7 @@ source "net/mpls/Kconfig"
 source "net/hsr/Kconfig"
 
 config RPS
-	boolean
+	boolean "RPS"
 	depends on SMP && SYSFS
 	default y
 
-- 
1.9.1

