From 2a55e9b989b558bed72a688da22460afa8ff0a6f Mon Sep 17 00:00:00 2001
From: Dmitry Shmidt <dimitrysh@google.com>
Date: Thu, 3 Jun 2010 10:55:33 -0700
Subject: [PATCH 085/634] network: wireless: Add get_mac_addr functionality to
 platform

Signed-off-by: Dmitry Shmidt <dimitrysh@google.com>
---
 include/linux/wlan_plat.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/wlan_plat.h b/include/linux/wlan_plat.h
index 70ee63b..3b1e2e0 100644
--- a/include/linux/wlan_plat.h
+++ b/include/linux/wlan_plat.h
@@ -20,6 +20,7 @@ struct wifi_platform_data {
 	int (*set_reset)(int val);
 	int (*set_carddetect)(int val);
 	void *(*mem_prealloc)(int section, unsigned long size);
+	int (*get_mac_addr)(unsigned char *buf);
 };
 
 #endif
-- 
1.9.1

