From cec693f3de8a3ffdbb0b566e6d8349f9215d6f27 Mon Sep 17 00:00:00 2001
From: Dmitry Shmidt <dimitrysh@google.com>
Date: Mon, 14 Feb 2011 16:58:48 -0800
Subject: [PATCH 086/634] net: wireless: Add get_country_code functionality to
 platform

Signed-off-by: Dmitry Shmidt <dimitrysh@google.com>
---
 include/linux/wlan_plat.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/wlan_plat.h b/include/linux/wlan_plat.h
index 3b1e2e0..40ec348 100644
--- a/include/linux/wlan_plat.h
+++ b/include/linux/wlan_plat.h
@@ -21,6 +21,7 @@ struct wifi_platform_data {
 	int (*set_carddetect)(int val);
 	void *(*mem_prealloc)(int section, unsigned long size);
 	int (*get_mac_addr)(unsigned char *buf);
+	void *(*get_country_code)(char *ccode);
 };
 
 #endif
-- 
1.9.1

