From f1b9f7243928291e703cf70ec5b2cc9cdc544121 Mon Sep 17 00:00:00 2001
From: Chia-chi Yeh <chiachi@android.com>
Date: Fri, 15 Jul 2011 15:32:57 -0700
Subject: [PATCH 095/634] net: Only NET_ADMIN is allowed to fully control TUN
 interfaces.

Signed-off-by: Chia-chi Yeh <chiachi@android.com>
---
 drivers/net/tun.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index ec63314..c789ec3 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1895,6 +1895,12 @@ static long __tun_chr_ioctl(struct file *file, unsigned int cmd,
 	unsigned int ifindex;
 	int ret;
 
+#ifdef CONFIG_ANDROID_PARANOID_NETWORK
+	if (cmd != TUNGETIFF && !capable(CAP_NET_ADMIN)) {
+		return -EPERM;
+	}
+#endif
+
 	if (cmd == TUNSETIFF || cmd == TUNSETQUEUE || _IOC_TYPE(cmd) == 0x89) {
 		if (copy_from_user(&ifr, argp, ifreq_len))
 			return -EFAULT;
-- 
1.9.1

