From 9482de4b00782ddcb2bf7afbe49a116805ab9d30 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Arve=20Hj=C3=B8nnev=C3=A5g?= <arve@android.com>
Date: Mon, 26 Nov 2012 16:09:13 -0800
Subject: [PATCH 250/634] usb: otg: otg-wakelock: Fix build for 3.7
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Arve Hjønnevåg <arve@android.com>
---
 drivers/usb/phy/otg-wakelock.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/phy/otg-wakelock.c b/drivers/usb/phy/otg-wakelock.c
index e17e272..479376b 100644
--- a/drivers/usb/phy/otg-wakelock.c
+++ b/drivers/usb/phy/otg-wakelock.c
@@ -16,6 +16,7 @@
 
 #include <linux/kernel.h>
 #include <linux/device.h>
+#include <linux/err.h>
 #include <linux/module.h>
 #include <linux/notifier.h>
 #include <linux/wakelock.h>
@@ -138,13 +139,15 @@ MODULE_PARM_DESC(enabled, "enable wakelock when VBUS present");
 static int __init otg_wakelock_init(void)
 {
 	int ret;
+	struct usb_phy *phy;
 
-	otgwl_xceiv = usb_get_transceiver();
+	phy = usb_get_phy(USB_PHY_TYPE_USB2);
 
-	if (!otgwl_xceiv) {
+	if (IS_ERR(phy)) {
 		pr_err("%s: No USB transceiver found\n", __func__);
-		return -ENODEV;
+		return PTR_ERR(phy);
 	}
+	otgwl_xceiv = phy;
 
 	snprintf(vbus_lock.name, sizeof(vbus_lock.name), "vbus-%s",
 		 dev_name(otgwl_xceiv->dev));
-- 
1.9.1

