From ed01c722efb757867922fa19a4b2f1a91ac85e3b Mon Sep 17 00:00:00 2001
From: JP Abgrall <jpa@google.com>
Date: Wed, 5 Feb 2014 12:29:34 -0800
Subject: [PATCH 328/634] ARM: tegra: flounder: stick to 32bit binder for now.

Signed-off-by: JP Abgrall <jpa@google.com>
---
 drivers/staging/android/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/android/Kconfig b/drivers/staging/android/Kconfig
index d1207b4..02703dd 100644
--- a/drivers/staging/android/Kconfig
+++ b/drivers/staging/android/Kconfig
@@ -22,6 +22,7 @@ config ANDROID_BINDER_IPC
 
 config ANDROID_BINDER_IPC_32BIT
 	bool "Use old 32-bit binder api"
+        default y
 	depends on !64BIT
 	---help---
 	  Enable to support an old 32-bit Android user-space. Breaks the new
-- 
1.9.1

