From e6a023aca8e32f9bfac8b68e8d112efb6d66c5ea Mon Sep 17 00:00:00 2001
From: Daniel Rosenberg <drosen@google.com>
Date: Mon, 12 Jan 2015 14:21:52 -0800
Subject: [PATCH 506/634] include: linux: cgroup: Fix compiler warning

Signed-off-by: Daniel Rosenberg <drosen@google.com>
---
 include/linux/cgroup.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/cgroup.h b/include/linux/cgroup.h
index ffd563d..1edbda9 100644
--- a/include/linux/cgroup.h
+++ b/include/linux/cgroup.h
@@ -883,7 +883,7 @@ static inline int cgroup_attach_task_all(struct task_struct *from,
 }
 
 static inline int subsys_cgroup_allow_attach(struct cgroup_subsys_state *css,
-					     struct cgroup_taskset *tset)
+					     void *tset)
 {
 	return 0;
 }
-- 
1.9.1

