From 504e290e892362f9e15121fa0b3a9df71b2096b2 Mon Sep 17 00:00:00 2001
From: Fei Yang <fei.yang@intel.com>
Date: Fri, 14 Nov 2014 11:51:41 -0800
Subject: [PATCH] ASoC:Intel: Potential list corruption.

Signed-off-by: Fei Yang <fei.yang@intel.com>
---
 sound/soc/intel/sst/sst.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/sound/soc/intel/sst/sst.c b/sound/soc/intel/sst/sst.c
index c112ff2..253a484 100644
--- a/sound/soc/intel/sst/sst.c
+++ b/sound/soc/intel/sst/sst.c
@@ -191,12 +191,15 @@ static irqreturn_t intel_sst_irq_thread_mfld(int irq, void *context)
 	struct ipc_post *__msg, *msg = NULL;
 	unsigned long irq_flags;
 
-	if (list_empty(&drv->rx_list))
+	if (drv->sst_state == SST_RECOVERY)
 		return IRQ_HANDLED;
 
 	spin_lock_irqsave(&drv->rx_msg_lock, irq_flags);
+	if (list_empty(&drv->rx_list)) {
+		spin_unlock_irqrestore(&drv->rx_msg_lock, irq_flags);
+		return IRQ_HANDLED;
+	}
 	list_for_each_entry_safe(msg, __msg, &drv->rx_list, node) {
-
 		list_del(&msg->node);
 		spin_unlock_irqrestore(&drv->rx_msg_lock, irq_flags);
 		if (msg->is_process_reply)
-- 
1.9.1

