From d4ef91a6ab124c7d4095cb4585cc417b8cce657a Mon Sep 17 00:00:00 2001
From: mark gross <mark.gross@intel.com>
Date: Tue, 26 Aug 2014 09:50:59 -0700
Subject: [PATCH 0359/1219] enable-32bit-binder-config-option

---
 drivers/staging/android/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/android/Kconfig b/drivers/staging/android/Kconfig
index 02703dd..22ac704 100644
--- a/drivers/staging/android/Kconfig
+++ b/drivers/staging/android/Kconfig
@@ -23,7 +23,7 @@ config ANDROID_BINDER_IPC
 config ANDROID_BINDER_IPC_32BIT
 	bool "Use old 32-bit binder api"
         default y
-	depends on !64BIT
+	depends on ANDROID_BINDER_IPC
 	---help---
 	  Enable to support an old 32-bit Android user-space. Breaks the new
 	  Android user-space.
-- 
1.9.1

