From a294f35c07b89f0b7636678c9970bd5220fd4d70 Mon Sep 17 00:00:00 2001
From: Scott D Phillips <scott.d.phillips@intel.com>
Date: Fri, 20 Jun 2014 23:56:55 +0000
Subject: [PATCH] netfilter: xt_qtaguid: Don't bother looking up TIME_WAIT sk's

Signed-off-by: Scott D Phillips <scott.d.phillips@intel.com>
---
 net/netfilter/xt_qtaguid.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/net/netfilter/xt_qtaguid.c b/net/netfilter/xt_qtaguid.c
index 0119b0d..5c97969 100644
--- a/net/netfilter/xt_qtaguid.c
+++ b/net/netfilter/xt_qtaguid.c
@@ -1695,9 +1695,10 @@ static bool qtaguid_mt(const struct sk_buff *skb, struct xt_action_param *par)
 	 * "struct inet_timewait_sock" which is missing fields.
 	 * So we ignore it.
 	 */
-	if (sk && sk->sk_state == TCP_TIME_WAIT)
+	if (sk && sk->sk_state == TCP_TIME_WAIT) {
 		sk = NULL;
-	if (sk == NULL) {
+		atomic64_inc(&qtu_events.match_found_no_sk_in_ct);
+	} else if (sk == NULL) {
 		/*
 		 * A missing sk->sk_socket happens when packets are in-flight
 		 * and the matching socket is already closed and gone.
-- 
1.7.9.5

