From: David Cohen <david.a.cohen@intel.com>
Subject: [PATCH] usb: phy-genetic: REMOVEME: force GPIO reset to be active low

Signed-off-by: David Cohen <david.a.cohen@intel.com>
---
Index: b/drivers/usb/phy/phy-generic.c
===================================================================
--- a/drivers/usb/phy/phy-generic.c	2014-06-25 10:58:15.912066390 -0700
+++ b/drivers/usb/phy/phy-generic.c	2014-06-25 10:59:54.351872058 -0700
@@ -83,7 +83,7 @@
 
 static inline void nop_reset_set(struct usb_phy_gen_xceiv *nop, int asserted)
 {
-	nop_func_set(nop, nop->gpio_reset, asserted, !asserted);
+	nop_func_set(nop, nop->gpio_reset, !asserted, !asserted);
 }
 
 static inline void nop_cs_set(struct usb_phy_gen_xceiv *nop, int asserted)
